metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/BaselineRuleEvaluator.java

```diff
@@ -47,43 +43,31 @@ public BaselineRuleEvaluator(QueryRequestHandler queryRequestHandler) {
             .get(0)
             .getBaselineThresholdCondition();
 
-    Duration duration = java.time.Duration.parse(dynamicThresholdCondition.getBaselineDuration());
-    long durationMillis = duration.toMillis();
+    long baselineDurationMillis =
+        java.time.Duration.parse(dynamicThresholdCondition.getBaselineDuration()).toMillis();
+    long ruleEvaluationStartTime =
+        alertTask.getCurrentExecutionTime()
+            - java.time.Duration.parse(metricAnomalyEventCondition.getRuleDuration()).toMillis();
 
     // this query will fetch metric data, which will used for baseline calculation and evaluation
-    QueryRequest queryRequest =
-        queryRequestHandler.getQueryRequest(
-            metricAnomalyEventCondition,
+    List<Pair<Long, Double>> dataList =
```

Comment:
nit: Can we just add a comment for readability for the Pair<Long, Double>. Mainly for its first component which represents (Long) timestamp and its unit is - millis ?